maint-lib/ceph_version.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+CEPH_VERSION_SPEC=$1
+WHAT_TO_EXTRACT=$2
+
+# Search for the two possible separator between CEPH_VERSION and CEPH_POINT_RELEASE
+# Let's consider CEPH_VERSION_SPEC=luminous-12.2.0-1
+for separator in "=" "-"; do
+  # If the line doesn't have a separator, let's try the next one
+  echo "${CEPH_VERSION_SPEC}" | grep -q "${separator}" || continue
+  case "${WHAT_TO_EXTRACT}" in
+    "CEPH_POINT_RELEASE")
+      # e.g Extracting -12.2.0-1 from CEPH_VERSION_SPEC
+      echo -n "${separator}"
+      echo "${CEPH_VERSION_SPEC}" | cut -d "${separator}" -f 2-
+      exit 0
+      ;;
+    *)
+      # If we don't ask for a CEPH_POINT_RELEASE it means that's CEPH_VERSION
+      # e.g Extracting luminous from CEPH_VERSION_SPEC
+      echo "${CEPH_VERSION_SPEC}" | cut -d "${separator}" -f 1
+      exit 0
+      ;;
+  esac
+done
+
+# If no separator got found, it means it doesn't have any
+# Let's consider CEPH_VERSION_SPEC=luminous
+case "${WHAT_TO_EXTRACT}" in
```

Comment:
There are 2 case statements which have the same test. This suggests that these 2 cases can be combined together. 

Another code-flow option would be to take the `CEPH_VERSION_SPEC` input and pull it apart into it's 2 parts initially with CEPH_POINT_RELEASE getting empty string if it can't be pulled apart. Then trivially output (echo) the desired part in a single case statement that comes after. Code will be DRYer.

Reply:
I simplified the bash script, hope you'll like it